test/functional/http/post/5.dapps.js

```diff
@@ -0,0 +1,503 @@
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+					goodTransactions.push(transaction);
+				});
+			});
+
+			it('with empty string should be ok', function () {
+				var application = node.randomApplication();
+				application.description = '';
+
+				transaction = node.lisk.dapp.createDapp(account.password, null, application);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+					goodTransactions.push(transaction);
+				});
+			});
+
+			it('with string longer than maximum(160) should fail', function () {
+				var application = node.randomApplication();
+				application.description = node.randomString.generate({
+					length: 161
+				});
+				transaction = node.lisk.dapp.createDapp(account.password, null, application);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.match(/String is too long \(161 chars\), maximum 160$/);
+					badTransactions.push(transaction);
+				});
+			});
+		});
+
+		describe('icon', function () {
+
+			it('without should be ok', function () {
+				var application = node.randomApplication();
+				delete application.icon;
+
+				transaction = node.lisk.dapp.createDapp(account.password, null, application);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+					goodTransactions.push(transaction);
+				});
+			});
+
+			it('with invalid url should fail', function () {
+				var application = node.randomApplication();
+				application.icon = 'invalidUrl';
+
+				transaction = node.lisk.dapp.createDapp(account.password, null, application);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.equal('Invalid application icon link');
+					badTransactions.push(transaction);
+				});
+			});
+
+			it('with invalid file type should fail', function () {
+				var application = node.randomApplication();
+				application.icon += '.invalid';
+
+				transaction = node.lisk.dapp.createDapp(account.password, null, application);
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.equal('Invalid application icon file type');
+					badTransactions.push(transaction);
+				});
+			});
+		});
+
+		describe('link', function () {
```

Comment:
Missing test for a non-string link
Missing test for missing link / undefined

Reply:
Same as before